File: 20190405腾讯常规批笔试/1.cpp
#include<iostream>
#include<vector>
#include<queue>
#include<map>
#include<list>
#include<set>
#include<algorithm>
using namespace std;

void test(vector<int>vec){
	int max = vec.size();
	int curmax = max;
	vector<int>v;
	for (int i = 0; i < vec.size(); ++i){
		v.push_back(vec[i]);
		if (vec[i] == curmax){
			sort(v.begin(), v.end());
			cout << v[v.size() - 1];
			v.erase(v.end() - 1);
			curmax--;
			while (v.size()>0 && v[v.size() - 1] == curmax){
				cout <<"s"<<v[v.size() - 1];
				v.erase(v.end() - 1);
				curmax--;
			}
		}
		 cout << endl;
	}
}
int main(){
	int t;
	cin >> t;
	vector<vector<int>>vec;
	for (int i = 0; i < t; ++i){
		int n;
		cin >> n;
		vector<int>v;
		for (int j = 0; j < n; ++j){
			int num;
			cin >> num;
			v.push_back(num);
		}
		vec.push_back(v);
	}
	for (int k = 0; k < vec.size(); ++k){
		test(vec[k]);
	}
	system("pause");
	return 0;
}


//int main(){
//	int n, k;
//	cin >> n >> k;
//	if (n < k)
//	{
//		cout << 0 << endl;
//	}
//	else
//	{
//		cout << (1 + (n - k)*(n - k)) * 2 << endl;
//	}
//	return 0;
//}

//int main(){
//	int t;
//	cin >> t;
//	for ()
//	system("pause");
//	return 0;
//}

File: 20190909奇安信/1.js
function findTree(set,map,num){
  if(map[num]){
      for(let j=0;j<map[num].length;++j){
          set.add(map[num][j]);
          findTree(set,map,map[num][j]);
      }
    }else{
        set.add(num);
    }
}

function test(arr1,arr2,num){
   let map={};
   for(let i=0;i<arr2.length;++i){
       if(!map[arr2[i]]){
           map[arr2[i]]=[arr1[i]];
       }else{
        map[arr2[i]].push(arr1[i]);
       }
   }
   if(!map[num])return 1;
   else{
    let set=new Set();
    findTree(set,map,num);
    console.log(set);
       return set.size+1;
   }
}

let arr1=[3,1,5,21,10];
let arr2=[0,3,3,1,5];
let num=3;
console.log(test(arr1,arr2,num));